src/navsections/sectionBlueprintMiddleware.ts

```diff
@@ -274,7 +274,9 @@ const processSectionBlueprints = async (state: RootState, dispatch: AppDispatch)
       visibleItemIds: g.itemIds.filter(itemId => itemInstanceMap[itemId].isVisible === true),
     }));
     const visibleItemIds = itemInstances?.filter(i => i.isVisible === true).map(i => i.id) || [];
-    const visibleGroupIds = sectionInstanceGroups.filter(g => g.visibleItemIds.length > 0).map(g => g.id);
+    const visibleGroupIds = sectionBlueprint.customization?.emptyGroupText
+      ? sectionInstanceGroups.map(g => g.id)
+      : sectionInstanceGroups.filter(g => g.visibleItemIds.length > 0).map(g => g.id);
```

Comment:
You can reduce this line by using reduce() method for this array, in that whay you will run only once through array

Reply:
I decided to go with a lodash sequence instead to keep the readability and also loop through the array only once